tests/unit/test_hnsw.cpp

```diff
@@ -1838,103 +1840,166 @@ TYPED_TEST(HNSWTest, rangeQueryCosine) {
+                         multilToString[i] + ".hnsw_v2";
 
-    // Create new index, set it into the serializer and extract the data to it.
-    auto new_index = test_utils::CreateNewIndex(params, VecSimType_FLOAT32);
-    ASSERT_EQ(VecSimIndex_IndexSize(new_index), 0);
-
-    serializer.reset(reinterpret_cast<HNSWIndex<float, float> *>(new_index));
-    serializer.loadIndex(file_name);
-
-    // Validate that the new loaded index has the same meta-data as the original.
-    VecSimIndexInfo new_info = VecSimIndex_Info(new_index);
-    ASSERT_EQ(info.algo, new_info.algo);
-    ASSERT_EQ(info.hnswInfo.M, new_info.hnswInfo.M);
-    ASSERT_EQ(info.hnswInfo.efConstruction, new_info.hnswInfo.efConstruction);
-    ASSERT_EQ(info.hnswInfo.efRuntime, new_info.hnswInfo.efRuntime);
-    ASSERT_EQ(info.hnswInfo.indexSize, new_info.hnswInfo.indexSize);
-    ASSERT_EQ(info.hnswInfo.max_level, new_info.hnswInfo.max_level);
-    ASSERT_EQ(info.hnswInfo.entrypoint, new_info.hnswInfo.entrypoint);
-    ASSERT_EQ(info.hnswInfo.metric, new_info.hnswInfo.metric);
-    ASSERT_EQ(info.hnswInfo.type, new_info.hnswInfo.type);
-    ASSERT_EQ(info.hnswInfo.dim, new_info.hnswInfo.dim);
-
-    res = serializer.checkIntegrity();
-    ASSERT_TRUE(res.valid_state);
-
-    // Add 1000 random vectors, override the existing ones to trigger deletions.
-    std::vector<float> data((n + 1) * dim);
-    std::mt19937 rng;
-    rng.seed(47);
-    std::uniform_real_distribution<> distrib;
-    for (size_t i = 0; i < (n + 1) * dim; ++i) {
-        data[i] = (float)distrib(rng);
-    }
-    for (size_t i = 0; i < n + 1; ++i) {
-        VecSimIndex_AddVector(new_index, data.data() + dim * i, i);
-    }
+        // Save the index with the default version (V2).
+        hnsw_index->saveIndex(file_name);
 
-    // Delete arbitrary vector (trigger removal of a block).
-    VecSimIndex_DeleteVector(new_index, (size_t)(distrib(rng) * (n + 1)));
+        // Fetch info after saving, as memory size change during saving.
+        VecSimIndexInfo info = VecSimIndex_Info(index);
+        ASSERT_EQ(info.algo, VecSimAlgo_HNSWLIB);
+        ASSERT_EQ(info.hnswInfo.M, M);
+        ASSERT_EQ(info.hnswInfo.efConstruction, ef);
+        ASSERT_EQ(info.hnswInfo.efRuntime, ef);
+        ASSERT_EQ(info.hnswInfo.indexSize, n);
+        ASSERT_EQ(info.hnswInfo.metric, VecSimMetric_L2);
+        ASSERT_EQ(info.hnswInfo.type, TypeParam::get_index_type());
+        ASSERT_EQ(info.hnswInfo.dim, dim);
+        ASSERT_EQ(info.hnswInfo.indexLabelCount, n_labels[i]);
 
-    HNSWIndex<float, float> *hnswNewIndex = reinterpret_cast<HNSWIndex<float, float> *>(new_index);
-    ASSERT_EQ(hnswNewIndex->getIndexCapacity(), n);
+        VecSimIndex_Free(index);
 
-    // Persist index, delete it from memory and restore.
-    serializer.saveIndex(file_name);
-    VecSimIndex_Free(new_index);
+        // Load the index from the file.
+        VecSimIndex *serialized_index = HNSWFactory::NewIndex(file_name, &params);
```

Comment:
Why sending `params`? I thought that for V2 you don't need them.

Reply:
remvoed